services/edit/queue.js

```diff
@@ -34,7 +34,7 @@ function add(fn, args, progressType) {
 
   // every time we add a function to the queue, check to see if it's already added
   if (queueCache[cacheHash]) {
-    return Promise.resolve(queueCache[cacheHash]());
+    return Promise.resolve(queueCache[cacheHash]);
```

Comment:
this means "resolve the promise that's currently pending", rather than the old one which, by calling the function again, actually resolved a new promise